### PR TITLE
fix: persisted query fix with fragments

### LINKED ISSUE #1021: Bug: Seems like the contributions list and notes reflect bdougie's account
### Describe the bug
When I pull up the site in production and go to one of my goal repos, contributions (issues) by bdougie show up instead of ones I've made. Similar thing for notes.

### To Reproduce
Someone besides bdougie pull up a goal for a repo you've both done contributions (and/or) notes for. I used FiniteSingularity/tau.

### Expected behavior
My issues / PRs would show up under Contributions.

### Screenshots
Screen Shot 2021-05-29 at 10 32 36 PM

### Desktop

- OS: Mac
- Browser: Chrome
- Version: latest

### Smartphone

- Device:
- OS:
- Browser:
- Version:

### PR DESCRIPTION
**Pull request by bdougie**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [ ] ✨ Feature
- [x] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description
There are two different things happening here. 

1. notes content is not rendering 
2. contributions are showing bdougie's contributions and not the user's

## Related Tickets & Documents

fixes #1021

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

Screen Shot 2021-06-16 at 10 57 24 PM

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [x] 📓 docs
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by bdougie** [comment]
This will need docs before being merged.

**Comment by bdougie** [comment]
@mtfoley this is fixed. When you have a second, could you check out FiniteSingularity/tau in the dashboard to see if my contribution shows.

**Comment by mtfoley** [comment]
@bdougie looks like Netlify deploy failed on this. Not a ton I could figure with Netlify logs on the smartphone but looks like `npm run build` failed, dealing with  apollo-boost.

**Comment by 0-vortex** [comment]
> @bdougie looks like Netlify deploy failed on this. Not a ton I could figure with Netlify logs on the smartphone but looks like `npm run build` failed, dealing with apollo-boost.

Fail on my end :< #1041

**Comment by mtfoley** [comment]
Pulled it up in safari mobile. Contributions looks right, notes doesn’t. Doesn’t render any notes and when I do edit notes, the text area still has @bdougie ’s notes.

**Comment by bdougie** [comment]
> Doesn’t render any notes and when I do edit notes, the text area still has @bdougie ’s notes.

Notes will need to be addressed in a different PR. That data comes from the authenticated GitHub App and not a persisted query. Currently working on docs to better explain this.

**Comment by 0-vortex** [comment]
> > Doesn’t render any notes and when I do edit notes, the text area still has @bdougie ’s notes.
> 
> It looks like that has been broken since this PR was merge. [53225e6](https://github.com/open-sauced/open-sauced/commit/53225e6af1cd6ee9209ff3e273b6b0496b554a57)
> 
> We went from v4 to v6 and the API is different. Instead of:
> 
> ```js
> <ReactMarkdown source={note}>
> ```
> 
> It now needs to be
> 
> ```js
> <ReactMarkdown children={note}>
> ```

Completely agree, dependabot should be used just for minor++ security issues, having all the dependencies cleaned up in `package.json` we can handle the major version bumps ourselves. In the React@17 update I missed checking packages I did not update. There are still some packages that could be updated:

```
 babel-jest               ^26.6.3  →  ^27.0.2
 jest                     ^26.6.3  →  ^27.0.4
 jest-axe                  ^4.1.0  →   ^5.0.1
 postcss-loader            ^4.3.0  →   ^6.1.0
 webpack                  ^4.41.5  →  ^5.39.0
 webpack-manifest-plugin   ^2.2.0  →   ^3.1.1
 graphql                  ^14.7.0  →  ^15.5.0
```

**Comment by mtfoley** [comment]
Renders notes now. Still notes from @bdougie’s goals repo rather than mine, but I’m assuming that’s a fix in persisted query.

**Comment by bdougie** [comment]
Confirmed on stream, this is grabbing bdougie's first note.

**Comment by 0-vortex** [comment]
From reading the docs it was my understanding that the fragments would cache appropriate data, however after discovering #1064 there is a possibility we have been running the project with missing dependencies, `apollo-boost` handling the cache. 

We should check whether some of our previous assessments regarding this issue on live the past couple of weeks were caused by that.

**Comment by bdougie** [comment]
Currently blocked on this error after this last commit. It has something to do with he fetch from the API, but I am not getting enough detail from the error message. 

Screen Shot 2021-07-13 at 4 29 44 PM


In the network tab, it states, "could not coerce number into an int." Perhaps something I am overlooking.

**Comment by mtfoley** [comment]
@bdougie You pull the issue number out of the router as "id" in that last change but then your run parseInt against "issueNumber". Could that be it?

File: src/components/Repository.js [hunk 1]
```diff
@@ -56,14 +56,12 @@ function Repository({user, match}) {
       });
 
     api
-      .fetchGoalQuery({number: parseInt(issueNumber)})
```
Comment (bdougie) [inline src/components/Repository.js]: ah, this maKes sense. Not meant to send an object here.

**Comment by bdougie** [comment]
This is interesting. There were a few moving parts in this feature in the last and I can think of a few ways why this broke. Thanks for the extra eyes (commits) here @0-vortex.

**Comment by 0-vortex** [comment]
> This is interesting. There were a few moving parts in this feature in the last and I can think of a few ways why this broke. Thanks for the extra eyes (commits) here @0-vortex.

@mtfoley kept me going ❤️

**Comment by bdougie** [comment]
@mtfoley when you have a moment (no rush), please review this and confirm the notes show up correctly for you. I think if it is good for you, then we can merge.

**Comment by mtfoley** [comment]
Stoked to say it works for me!

**Review by mtfoley — APPROVED** [review]
LGTM, this has been really fun to watch!

**Pull request merged**